File: perl/etc/justonce/giga.p.pl
#!/usr/bin/env perl
use strict;
use warnings;
use DBI;
use Data::Dumper;

my $dbh = DBI->connect("dbi:SQLite:dbname=giga.authors.sqlite","","",{RaiseError => 0,PrintError => 1,AutoCommit => 0}) or die $DBI::errstr;

my $sthi = $dbh->prepare("SELECT * FROM PubDat;");
$sthi->execute();
my $ret = $sthi->fetchall_arrayref;

if ($#$ret == -1) {
	die "[x]Empty !\n";
}

for (@$ret) {
	print "DOI: $_->[0]\n";
	print "Title: $_->[1]\n";
	print "Type: $_->[2]\n";
	print "Authors: { $_->[3] }\n";
	print "RefList: { $_->[4] }\n";
}

$dbh->rollback;
$dbh->disconnect;

File: python/etc/giga.p.py
#!/usr/bin/env python3

import os, sys, bz2, csv, re, json, sqlite3

conn = sqlite3.connect('giga.authors.sqlite')

for row in conn.execute("SELECT * FROM PubDat"):
    #print(row)
    auther=json.loads(row[3])
    print('DOI:%s\nTitle:%s\nType:%s\nRef:{%s}\nAuthers:'%(row[0],row[1],row[2],row[4]))
    print(auther)

